File: TTdsa-java/Exercise/src/com/aacirq/math/N47.java
package com.aacirq.math;

public class N47 {
    public static void main(String[] args) {
        N47 sol = new N47();

        System.out.println(sol.Sum_Solution(5));
    }

    private int res = 0;
    public int Sum_Solution(int n) {
        boolean b = (n > 0) && (Sum_Solution(n - 1) > 0);
        res += n;
        return res;
    }
}

File: TTdsa-cpp/Exercise/math/N47.cpp
//
// Created by renxin on 2021/6/21.
//

#include "ttdsa.h"

class Solution {
public:
    int res = 0;
    int Sum_Solution(int n) {
        bool b = (n > 0) && (Sum_Solution(n - 1) > 0);
        res += n;
        return res;
    }
};

int main() {
    Solution sol;
    cout << sol.Sum_Solution(5) << endl;
    return 0;
}